lib/straight/gateway.rb

```diff
@@ -90,32 +89,33 @@ def new_order(args)
         if args[:amount].nil? || !args[:amount].kind_of?(Numeric) || args[:amount] <= 0
           raise OrderAmountInvalid, "amount cannot be nil and should be more than 0"
         end
-        # Setting default values
-        args[:currency]         ||= default_currency
-        args[:btc_denomination] ||= :satoshi
 
-        amount = args[:amount_from_exchange_rate] = amount_from_exchange_rate(
+        # Build order with default values
+        order = Kernel.const_get(order_class).new
+        order.gateway = self
+        order.keychain_id = args[:keychain_id]
+        order.block_height_created_at = fetch_latest_block_height rescue nil
+
+         args[:currency] ||= default_currency
+
+        args[:amount_from_exchange_rate] = amount_from_exchange_rate(
           args[:amount],
-          currency:         args[:currency],
-          btc_denomination: args[:btc_denomination]
+          currency: args[:currency],
+          btc_denomination: args[:btc_denomination] || :satoshi,
         )
```

Comment:
> Do you consider bad that args changed or that it didn't?

I think that changing arguments inside function is bad (except setting default values).
> We could actually also remove the args[:amount_from_exchange_rate] intermediate assignment and assign it directly to order.amount.

Sure

For 2) Oh, it's my fault, sorry. I didn't see it.

> I have no problem going back to the previous version, although personally I find this version easier to read and elegant enough

Leave your variant with removing `args[:amount_from_exchange_rate] = ... ` and unnecessary one space in line 99.